Framework/Algorithms/src/CalculatePlaczekSelfScattering.cpp

```diff
@@ -0,0 +1,163 @@
+  const double neutron_mass = factor * 1.674927471e-27; // neutron mass
+  // get sample information : mass, total scattering length, and concentration
+  // of each species
+  auto atom_species =
+      CalculatePlaczekSelfScattering::get_sample_species_info(inWS);
+  // calculate summation term w/ neutron mass over molecular mass ratio
+  double summation_term = 0.0;
+  for (auto t = atom_species.begin(); t != atom_species.end(); t++) {
+    summation_term += t->second["concentration"] * t->second["b_sqrd_bar"] *
+                      neutron_mass / t->second["mass"];
+  }
+  // get incident spectrum and 1st derivative
+  const MantidVec x_lambda = inWS->readX(0);
+  const MantidVec incident = inWS->readY(0);
+  const MantidVec incident_prime = inWS->readY(1);
+  double dx = (x_lambda[1] - x_lambda[0]) / 2.0;
+  std::vector<double> phi_1;
+  for (size_t i = 0; i < x_lambda.size() - 1; i++) {
+    phi_1.push_back((x_lambda[i] + dx) * incident_prime[i] / incident[i]);
+  }
+  // set detector law term (eps_1)
+  const double lambda_D = 1.44;
+  std::vector<double> eps_1;
+  for (size_t i = 0; i < x_lambda.size(); i++) {
+    double x_term = -x_lambda[i] / lambda_D;
+    eps_1.push_back(x_term * exp(x_term) / (1.0 - exp(x_term)));
+  }
+  /* Placzek
+     Original Placzek inelastic correction Ref (for constant wavelength, reactor
+     source): Placzek, Phys. Rev v86, (1952), pp. 377-388 First Placzek
+     correction for time-of-flight, pulsed source (also shows reactor eqs.):
+     Powles, Mol. Phys., v6 (1973), pp.1325-1350
+     Nomenclature and calculation for this program follows Ref:
+     Howe, McGreevy, and Howells, J. Phys.: Condens. Matter v1, (1989), pp.
+     3433-3451 NOTE: Powles's Equation for inelastic self-scattering is equal to
+     Howe's Equation for P(theta) by adding the elastic self-scattering
+  */
+  MantidVec x_lambdas;
+  MantidVec placzek_correction;
+  const Geometry::DetectorInfo det_info = inWS->detectorInfo();
+  int NSpec = 0;
+  for (size_t det_index = 0; det_index < det_info.size(); det_index++) {
+    if (det_info.isMonitor(det_index) != true) {
+      NSpec += 1;
+      const double path_length = det_info.l1() + det_info.l2(det_index);
+      const double f = det_info.l1() / path_length;
+      const double sin_theta_by_2 = sin(det_info.twoTheta(det_index) / 2.0);
+      for (size_t x_index = 0; x_index < x_lambda.size() - 1; x_index++) {
+        const double term1 = (f - 1.0) * phi_1[x_index];
+        const double term2 = f * eps_1[x_index];
+        const double term3 = f - 3;
+        const double inelastic_placzek_self_correction =
+            2.0 * (term1 - term2 + term3) * sin_theta_by_2 * sin_theta_by_2 *
+            summation_term;
+        x_lambdas.push_back(x_lambda[x_index]);
+        placzek_correction.push_back(inelastic_placzek_self_correction);
+      }
+      x_lambdas.push_back(x_lambda.back());
+    }
+  }
+  Mantid::API::Algorithm_sptr ChildAlg =
```

Comment:
Variable names in this file are a mix of PascalCase and snake_case, our guidelines dictate canalCase.